cmd/services/main.go

```diff
@@ -114,6 +115,13 @@ func promoteAction(c *cli.Context) error {
 	newBranchName := c.String(branchNameFlag)
 	debug := c.Bool(debugFlag)
 
+	if newBranchName == "" {
+		branch, err := exec.Command("uuidgen").Output()
```

Comment:
Might it be possible to do this without shelling out? We have `uuidgen` on Macs, but would have to be careful to install it into any docker image we baked this into. See https://blog.kowalczyk.info/article/JyRZ/generating-good-unique-ids-in-go.html for some options. Also please can we have a test that confirms that a (large-ish, perhaps a thousand?) branch names are all different?

Reply:
If you use uuid generation, it would be a big problem if the UUIDs weren't different ;-)

Reply:
@bigkevmcd 's suggestion of using `git rev-parse --short HEAD` as a prefix is better though: it will make it a bit more obvious what the target branch relates to.